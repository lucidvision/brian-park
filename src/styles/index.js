export const colors = {
  primary: 'rgba(0, 13, 85, 0.8)',
  black: '#000',
  white: '#fff',
  gradientLight: 'rgba(0, 38, 255, 0.8)',
  gradientDark: 'rgba(0, 0, 0, 0.8)'
}

export const fontSizes = {
  large: '6rem',
  medium: '4rem',
  small: '2.2rem',
  default: '1.6rem'
}

export const screenSizes = {
  desktop: '112.5em',
  tablet: '75em',
  phone: '56.25em'
}

export const utilities = {
  marginSmall: '1.5rem',
  marginMedium: '4rem',
  marginLarge: '8rem'
}
