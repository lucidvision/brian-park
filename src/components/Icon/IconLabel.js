import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { HeadingTertiary } from 'components'
import { colors } from 'styles'

IconLabel.propTypes = {
  className: PropTypes.string.isRequired,
  color: PropTypes.string,
  label: PropTypes.string.isRequired
}

IconLabel.defaultProps = {
  className: '',
  color: colors.primary,
  label: ''
}

const StyledSpan = styled.span`
  text-align: center;
`

const StyledIcon = styled.i`
  font-size: 6em;
  color: ${props => props.color}
`

export default function IconLabel (props) {
  return (
    <StyledSpan>
      <StyledIcon className={props.className} {...props} />
      <HeadingTertiary color={props.color} text={props.label} />
    </StyledSpan>
  )
}
