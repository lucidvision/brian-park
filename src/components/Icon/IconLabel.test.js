import React from 'react'
import { mount } from 'enzyme'
import IconLabel from './IconLabel'
import { colors } from 'styles'

const createIconLabel = (override) => {
  const props = {
    label: 'Dolphins',
    ...override
  }

  return mount(<IconLabel {...props} />)
}

describe('IconLabel', () => {
  const component = createIconLabel()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders with the correct props', () => {
    expect(component.find('IconLabel').props()).toEqual({
      color: colors.primary,
      label: 'Dolphins'
    })
  })

  it('renders the icon', () => {
    expect(component.find('i').exists()).toBeTruthy()
  })

  it('renders the tertiary heading with the correct props', () => {
    expect(component.find('HeadingTertiary').props()).toEqual({
      color: colors.primary,
      text: 'Dolphins'
    })
  })
})
