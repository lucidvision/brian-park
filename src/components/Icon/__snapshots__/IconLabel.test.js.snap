// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLabel renders correctly 1`] = `
<IconLabel
  color="rgba(0, 13, 85, 0.8)"
  label="Dolphins"
>
  <styled.span>
    <span
      className="sc-gZMcBi ibDgNx"
    >
      <styled.i>
        <i
          className="sc-gqjmRU dfnTHc"
        />
      </styled.i>
      <HeadingTertiary
        color="rgba(0, 13, 85, 0.8)"
        text="Dolphins"
      >
        <styled.h3
          color="rgba(0, 13, 85, 0.8)"
        >
          <h3
            className="sc-bZQynM jZhXzy"
            color="rgba(0, 13, 85, 0.8)"
          >
            Dolphins
          </h3>
        </styled.h3>
      </HeadingTertiary>
    </span>
  </styled.span>
</IconLabel>
`;
